engines/io_uring.c

```diff
@@ -554,6 +554,46 @@ static struct io_u *fio_ioring_cmd_event(struct thread_data *td, int event)
 	return io_u;
 }
 
+static char *fio_ioring_cmd_errdetails(struct io_u *io_u)
+{
+	unsigned int sct = (io_u->error >> 8) & 0x7;
+	unsigned int sc = io_u->error & 0xff;
+#define MAXERRDETAIL 1024
+#define MAXMSGCHUNK 128
+	char *msg, msgchunk[MAXMSGCHUNK];
+	int ctrlid, nsid;
+	int ret;
+
+	msg = calloc(1, MAXERRDETAIL);
+	strcpy(msg, "io_uring_cmd: ");
+
+	snprintf(msgchunk, MAXMSGCHUNK, "%s: ", io_u->file->file_name);
+	strlcat(msg, msgchunk, MAXERRDETAIL);
+
+	/*
+	 * Check the filename to figure out cmd_type since we don't have
+	 * thread_data here.
+	 */
+	ret = sscanf(io_u->file->file_name, "/dev/ng%dn%d", &ctrlid, &nsid);
```

Comment:
Can you just use `strstr` to check if the filename begins with `/dev/ng`? Then there is no need to have the `ctrlid` and `nsid` variables which are ultimately unused.

Reply:
Okay!